pkg/addons/default/coredns.go

```diff
@@ -167,54 +162,81 @@ func InstallCoreDNS(rawClient kubernetes.RawClientInterface, region string, wait
 	return false, nil
 }
 
-// UpdateCoreDNSImageTag updates image tag for kube-system:deployment/coredns based to match the latest release
-func UpdateCoreDNSImageTag(clientSet k8s.Interface, plan bool) (bool, error) {
-	printer := printers.NewJSONPrinter()
+// UpdateCoreDNS will update the `coredns` add-on
+func UpdateCoreDNS(rawClient kubernetes.RawClientInterface, region, controlPlaneVersion string, plan bool) (bool, error) {
+	kubeDNSSevice, err := rawClient.ClientSet().CoreV1().Services(metav1.NamespaceSystem).Get(KubeDNS, metav1.GetOptions{})
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			logger.Warning("%q service was not found", KubeDNS)
+			return false, nil
+		}
+		return false, errors.Wrapf(err, "getting %q service", KubeDNS)
+	}
 
-	d, err := clientSet.AppsV1().Deployments(metav1.NamespaceSystem).Get(CoreDNS, metav1.GetOptions{})
+	_, err = rawClient.ClientSet().AppsV1().Deployments(metav1.NamespaceSystem).Get(CoreDNS, metav1.GetOptions{})
 	if err != nil {
 		if apierrs.IsNotFound(err) {
 			logger.Warning("%q was not found", CoreDNS)
 			return false, nil
 		}
 		return false, errors.Wrapf(err, "getting %q", CoreDNS)
 	}
-	if numContainers := len(d.Spec.Template.Spec.Containers); !(numContainers >= 1) {
-		return false, fmt.Errorf("%s has %d containers, expected at least 1", CoreDNS, numContainers)
-	}
 
-	if err := printer.LogObj(logger.Debug, CoreDNS+" [current] = \\\n%s\n", d); err != nil {
+	// if Deployment is present, go through our list of assets
+	list, err := loadAssetCoreDNS(controlPlaneVersion)
+	if err != nil {
 		return false, err
 	}
 
-	image := &d.Spec.Template.Spec.Containers[0].Image
-	imageParts := strings.Split(*image, ":")
+	for _, rawObj := range list.Items {
+		resource, err := rawClient.NewRawResource(rawObj)
+		if err != nil {
+			return false, err
+		}
+		switch resource.GVK.Kind {
+		case "Deployment":
+			image := &resource.Info.Object.(*appsv1.Deployment).Spec.Template.Spec.Containers[0].Image
```

Comment:
Is this ok `...Containers[0]`? Do we know it will always have one item?

Reply:
Yes, because it's defined by the JSON manifest that we control.